src/parquet/file/metadata.cc

```diff
@@ -474,29 +475,33 @@ void FileMetaData::WriteTo(OutputStream* dst) {
 }
 
 ApplicationVersion::ApplicationVersion(const std::string& created_by) {
-  namespace ba = boost::algorithm;
+  std::regex app_regex(ApplicationVersion::APPLICATION_FORMAT);
+  std::regex ver_regex(ApplicationVersion::VERSION_FORMAT);
+  std::smatch app_matches;
+  std::smatch ver_matches;
 
   std::string created_by_lower = created_by;
   std::transform(created_by_lower.begin(), created_by_lower.end(),
       created_by_lower.begin(), ::tolower);
 
-  std::vector<std::string> tokens;
-  ba::split(tokens, created_by_lower, ba::is_any_of(" "), ba::token_compress_on);
-  // Boost always creates at least one token
-  DCHECK_GT(tokens.size(), 0);
-  application = tokens[0];
-
-  if (tokens.size() >= 3 && tokens[1] == "version") {
-    std::string version_string = tokens[2];
-    // Ignore any trailing nodextra characters
-    int n = version_string.find_first_not_of("0123456789.");
-    std::string version_string_trimmed = version_string.substr(0, n);
-
-    std::vector<std::string> version_tokens;
-    ba::split(version_tokens, version_string_trimmed, ba::is_any_of("."));
-    version.major = version_tokens.size() >= 1 ? atoi(version_tokens[0].c_str()) : 0;
-    version.minor = version_tokens.size() >= 2 ? atoi(version_tokens[1].c_str()) : 0;
-    version.patch = version_tokens.size() >= 3 ? atoi(version_tokens[2].c_str()) : 0;
+  bool app_success = std::regex_match(created_by_lower, app_matches, app_regex);
+  bool ver_success = false;
+
+  if (app_success && app_matches.size() >= 4) {
+    // first match is the entire string. sub-matches start from second.
+    application_ = app_matches[1];
```

Comment:
Appears `application_` is only being set in this branch, needs to be set to `"unknown"` otherwise